plugins/3scale-backend/README.md

```diff
@@ -34,32 +34,105 @@ yarn workspace backend add @janus-idp/backstage-plugin-3scale-backend
+       /* highlight-add-start */
+       builder.addEntityProvider(
+         ThreeScaleApiEntityProvider.fromConfig(env.config, {
+           logger: env.logger,
+           scheduler: env.scheduler,
+         }),
+       );
+       /* highlight-add-end */
+
+       const { processingEngine, router } = await builder.build();
+       await processingEngine.start();
+       return router;
+     }
+     ```
+
+     ***
+
+     **NOTE**
+
+     If you have made any changes to the schedule in the `app-config.yaml` file, then restart to apply the changes.
+
+     ***
+
+   - **Method 2**: Add a schedule directly inside the `packages/backend/src/plugins/catalog.ts` file as follows:
+
+     ```ts title="packages/backend/src/plugins/catalog.ts"
+     /* highlight-add-next-line */
+     import { ThreeScaleApiEntityProvider } from '@janus-idp/backstage-plugin-3scale-backend';
+
+     export default async function createPlugin(
+       env: PluginEnvironment,
+     ): Promise<Router> {
+       const builder = await CatalogBuilder.create(env);
+
+       /* ... other processors and/or providers ... */
+       /* highlight-add-start */
+       builder.addEntityProvider(
+         ThreeScaleApiEntityProvider.fromConfig(env.config, {
+           logger: env.logger,
+           schedule: env.scheduler.createScheduledTaskRunner({
+             frequency: { minutes: 1 },
+             timeout: { minutes: 1 },
+           }),
+         }),
+       );
+       /* highlight-add-end */
+
+       const { processingEngine, router } = await builder.build();
+       await processingEngine.start();
+       return router;
+     }
+     ```
+
+   - If both the `schedule` (hard-coded schedule) and `scheduler` (`app-config.yaml` schedule) option are provided in the `packages/backend/src/plugins/catalog.ts`, the `scheduler` option takes precedence.
+
+     - If the schedule inside the `app-config.yaml` file is not configured, then the `schedule` option is used.
+
+     ```ts title="packages/backend/src/plugins/catalog.ts"
+     /* highlight-add-next-line */
+     import { ThreeScaleApiEntityProvider } from '@janus-idp/backstage-plugin-3scale-backend';
+
+     export default async function createPlugin(
+       env: PluginEnvironment,
+     ): Promise<Router> {
+       const builder = await CatalogBuilder.create(env);
+
+       /* ... other processors and/or providers ... */
+       /* highlight-add-start */
+       builder.addEntityProvider(
+         ThreeScaleApiEntityProvider.fromConfig(env.config, {
+           logger: env.logger,
+           schedule: env.scheduler.createScheduledTaskRunner({
+             frequency: { minutes: 1 },
+             timeout: { minutes: 1 },
+           }),
+           scheduler: env.scheduler,
+         }),
+       );
+       /* highlight-add-end */
+
+       const { processingEngine, router } = await builder.build();
+       await processingEngine.start();
+       return router;
+     }
+     ```
```

Comment:
Isn't this just a repetition?

Besides, having both `scheduler` and `schedule` defined is an antipattern that should not be encouraged.

Reply:
Ah I did not know this was an antipattern. I assumed the `schedule` would serve as a hard coded default while the `scheduler` served as something the user can customize to override the hard coded default.

Reply:
I can remove the repeated code for when both `scheduler` and `schedule` are defined.

Reply:
Oh.. I didn't think of that... Maybe we're making it a bigger issue than it is. I would say the user should configure it in one or the other place and not both to keep it simple. The reason why there are 2 ways is that in typescript user has more options on how to do context-aware configuration if they need so (like conditionals, feature flag checks etc...) In most cases the `schedule` in `app-config.yaml` should be sufficient and preferred way.

Reply:
Ok, so should I just leave the behavior when both are defined as a mention, but just mention it's not recommended?